src/lib/github/apply-roles.ts

```diff
@@ -0,0 +1,451 @@
+// false otherwise or if error
+// (uses access token to determine current user)
+async function isOrgMember(accessToken: string, ghUserId: string) {
+  const octokit = new Octokit({
+    auth: `token ${accessToken}`,
+  })
+  try {
+    const { data } = await octokit.request('GET /orgs/{org}/members', {
+      org: process.env.GITHUB_ORG_LOGIN,
+    })
+    const index = data.findIndex(
+      (contributor) => contributor.id === Number(ghUserId)
+    )
+    if (index !== -1) return true
+  } catch (err) {
+    console.error(
+      `Failed to find org member in ${process.env.GITHUB_ORG_LOGIN}: ${err.response.data.message}`
+    )
+  }
+  return false
+}
+
+// returns a list of the given organization's repositories,
+// false if error
+// also this will only work if repos are public
+export async function fetchOrgRepos(accessToken) {
+  const octokit = new Octokit({
+    auth: `token ${accessToken}`,
+  })
+  try {
+    const { data } = await octokit.request('GET /orgs/{org}/repos', {
+      org: process.env.GITHUB_ORG_LOGIN,
+    })
+    return data
+  } catch (err) {
+    console.error(
+      `Failed to fetch repos for ${process.env.GITHUB_ORG_LOGIN}: ${err.response.data.message}`
+    )
+    return false
+  }
+}
+
+// for each repository belonging to the org, retrieves a list of
+// contributors. returns true if the user with a given id is
+// a contributor in at least one repository,
+// false otherwise or if error
+export async function isContributor(
+  accessToken: string,
+  repos: [],
+  userId: string
+) {
+  const octokit = new Octokit({
+    auth: `token ${accessToken}`,
+  })
+
+  for (let i = 0; i < repos.length; i++) {
+    const amplifyRepo = repos[i].name
+
+    try {
+      const { data } = await octokit.request(
+        'GET /repos/{owner}/{repo}/contributors',
+        {
+          owner: process.env.GITHUB_ORG_LOGIN,
+          repo: amplifyRepo,
+        }
+      )
+
+      const index = data.findIndex(
+        (contributor) => contributor.id === Number(userId)
+      )
+      if (index !== -1) return true
+    } catch (err) {
+      console.error(
+        `Error searching for user in repository ${amplifyRepo}: ${err.response.data.message}`
+      )
+    }
+  }
+  return false
+}
+
+// driver code that checks github membership/contribution status and applies roles
+export async function appplyRoles(
+  userId: string,
+  ghUserId: string,
+  accessToken: string
+) {
+  let discUserId
+  let staffResponse = true
+  let contributorResponse = true
+
+  // retrieve discord user id for current user
+  const data = await prisma.user.findFirst({
+    where: {
+      id: userId,
+    },
+    select: {
+      accounts: {
+        where: {
+          provider: 'discord',
+        },
+      },
+    },
+  })
+  if (data && data?.accounts && data?.accounts?.length === 1) {
+    discUserId = data.accounts[0].providerAccountId
+  }
+
+  if (!accessToken || !ghUserId || !discUserId) return false
```

Comment:
since accessToken and ghUserId are required parameters we don't need this additional check